### ISSUE TITLE
directly query LDAP in ocis-reva

### ISSUE THREAD

**Issue by synnove** [owncloud/ocis-reva#331]
Currently ocis-reva appears to need nscd/nslcd running alongside it to query LDAP for user lookups. It would be interesting to try doing queries directly using a library instead.

**Comment by butonic** [comment]
yes, currently the eos driver uses the go stdlib to look up the numeric userid. AFAICT it is only needed to make the eos call with a numeric user and group id ...

since this is an implementation detail of the storage driver I'd love to leave the aspect out of cs3, however the storage needs to be able to be integrated with all users (including guest acounts). ocis uses the ocis-glauth to provide an ldap api with numeric ids for all users (including guests). It can be replaced with a dedicated ldap server, but it then needs to contain guest accounts as well. otherwise sharing with guest accounts is not possible.

I think the numeric id should not be part of the cs3 api, but we need to better document that there might be dependencies on ldap for storage drivers.

In any case making use of ldap (or the users service?) in the eos storage driver would allow running a single process in the eos storeage driver container ... that being said: caching user and group lookup using nslcd has its merits too ...

**Comment by butonic** [comment]
tracked in reva as https://github.com/cs3org/reva/issues/964
